NAS-110294 / 12.0 / net/samba - fix tevent_kqueue error handling edge case

anodos325 opened the pull request:
kevent() in parent process can return -1 without setting errno
in case we receive a signal. Refactor error handling into separate
function to slightly clarify what is happening.

kevent() call in loop_once() function is only place where this is
needed. Calls submitting new kevents for queue should return immediately
since nevents is zero.

anodos325 committed:
net/samba - fix tevent_kqueue error handling edge case

kevent() in parent process can return -1 without setting errno
in case we receive a signal. Refactor error handling into separate
function to slightly clarify what is happening.

kevent() call in loop_once() function is only place where this is
needed. Calls submitting new kevents for queue should return immediately
since nevents is zero.

bugclerk commented:
Jira URL: https://jira.ixsystems.com/browse/NAS-110294

rick-mesta commented on net/samba/files/tevent_kqueue.patch:
```diff
@@ -723,47 +789,27 @@ index 00000000000..c574c65eb1f
-+					  kqueue_ev->kev_array, kqueue_ev->kev_size, tsp);
++		nkevents = check_kevent_errors(kqueue_ev, tsp);
++		if (nkevents == -1) {
++			return -1;
```
**nit:** back in line 723, you return `nkevents`... consistency in coding style is good 😁

anodos325 replied:
Okay. Will fix. I'll double-check again tomorrow with fresh eyes and add more comments.

rick-mesta reviewed (approved):
Other than the nit (which you can feel free to ignore), this LGTM 👍🏼

anodos325 committed:
net/samba - add comments to tevent_kqueue

Add some comments to provide overview of certain important
functions in tevent_kqueue.

anodos325 committed:
net/samba tevent_kqueue - style fix
